interpreter.py

```diff
@@ -1,46 +1,68 @@
 from __future__ import with_statement
-
-import sys
-
 from rpython.rlib import jit
 from llvm_wrapper import *
+from type_wrapper import *
 from state import State
 
+import sys
+
 def target(*args):
     return main, None
 
 def jitpolicy(driver):
     from rpython.jit.codewriter.policy import JitPolicy
     return JitPolicy()
 
-jit_driver = jit.JitDriver (
-    greens = [],
-    reds = [])
+jit_driver = jit.JitDriver(greens=[], reds=[])
 
 class NoSuchVariableException(Exception):
     pass
 
+class NoSuchTypeException(Exception):
+    pass
+
+def set_var(local_vars, global_vars, var, new_value):
+    ''' Changes the value of an existing variable. '''
+
+    assert isinstance(new_value, Type)
```

Comment:
Duplicated casts here. You can compute the integer address once using a local variable.